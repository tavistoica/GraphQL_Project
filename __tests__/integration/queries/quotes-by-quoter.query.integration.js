const { createTestClient } = require("apollo-server-testing");
const { gql } = require("apollo-server");
const {
  database: { url },
} = require("../../../src/core/config/test.config");
const Quote = require("../../../src/entities/quote/quote.entity");
const { Server } = require("../../../src/bootstrap/server.bootstrap");
const { Database } = require("../../../src/bootstrap/database.bootstrap");
const { generateID } = require("../../../src/core/helpers/generate-id.helper");

const server = new Server();
const database = new Database();

function getQuery() {
  return gql`
    query getQuotesByQuoter {
      quotesByquoter(quoter: "mafiotu") {
        phrase
      }
    }
  `;
}

async function createContext() {
  const quote = await createQuote({
    id: generateID,
    phrase: "O vorba mare",
    quoter: "mafiotu",
  });
  return {
    getQuotesQuery: getQuery(),
    quote,
  };
}

function createQuote(quote) {
  const newQuote = new Quote(quote);
  return newQuote.save();
}

beforeEach(async () => {
  await database.bootstrap({ url });
  server.bootstrap();
});
afterEach(() => database.reset());

it("quotesByquoter Query use case Test", async () => {
  const { getQuotesQuery } = await createContext();
  const { query } = createTestClient(server.apolloClient);
  const res = await query({
    query: getQuotesQuery,
  });
  expect(res).toMatchSnapshot();
});
