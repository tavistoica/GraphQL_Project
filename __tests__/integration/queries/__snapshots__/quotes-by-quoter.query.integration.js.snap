// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quotesByquoter Query use case Test 1`] = `
Object {
  "data": Object {
    "quotesByQuoter": Array [
      Object {
        "phrase": "If you can't explain it to a six year old, you don't understand it yourself.",
      },
      Object {
        "phrase": "If you can't explain it to a six year old, you don't understand it yourself.",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
